code-challenges.test.js

```diff
@@ -16,47 +16,162 @@
+// Create a variable called wordsContainingLetter to hold the new value 
+// Use the higher-order function .filter() that will loop through the array and return a subset of the array with only values that are truthy.
+// Use .include() method on the return value to find if the given variable letter is in the string
+// and use the .uppercase() method to make sure capitalized words get found
+
+
+// ----- Syntax -----
+const findCommonLetter = (array, letter) => {
+    return wordsContainingLetter = array.filter(value => {
+        // I have to include the .toUpperCase() method here, otherwise 'Apple' doesn't get found or returned
+        return value.includes(letter) || value.includes(letter.toUpperCase())
+    })
+}
+
+// Test passes
+
 
 
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
 
-var hand1 = [5, 5, 5, 3, 3]
+// var hand1 = [5, 5, 5, 3, 3]
 // Expected output: true
-var hand2 = [5, 5, 3, 3, 4]
+// var hand2 = [5, 5, 3, 3, 4]
 // Expected output: false
-var hand3 = [5, 5, 5, 5, 4]
+// var hand3 = [5, 5, 5, 5, 4]
 // Expected output: false
 
-
+// ----- Problem Solving -----
+// Create a describe method called findCommonLetter
+// Add a test method that describes what findCommonLetter does
+// Add an expect method that compares the input to the expected values
+
+// ----- Syntax -----
+describe("fullHouse", () => {
+    var hand1 = [5, 5, 5, 3, 3]
+    var hand2 = [5, 5, 3, 3, 4]
+    var hand3 = [5, 5, 5, 5, 4]
+    it("takes in an array of 5 numbers and determines whether or not the array is a full house (exactly one pair and one three of a kind)", () => {
+        expect(fullHouse(hand1)).toEqual(true)
+        expect(fullHouse(hand2)).toEqual(false)
+        expect(fullHouse(hand3)).toEqual(false)
+    })
+})
+
+// Test fails with ReferenceError: fullHouse is not defined
 
 // b) Create the function that makes the test pass.
+
+// ----- Problem Solving -----
+// A full house is an array of 2 of the same number AND 3 of the same number.
+// Iterate through the array to make sure there's no more than 2 of the same number.
+// If there's more than 2 of the same number, then return false. Probably be using the .length() method.
+// Create a function called fullHouse that takes in an array as a parameter.
+// Create a variable that holds the new value of the array
+
+
+// ----- Syntax -----
+const fullHouse = (array) => {
+    let countObj = {}
+    for(let x of array){
+      countObj[x] = (countObj[x] || 0) + 1;
+    }
+
+    let vals = Object.values(countObj);
+    if((vals[0] === 2 && vals[1] === 3) || (vals[1] === 2 && vals[0] === 3)){
+      return true;
+    }
+    return false;
+}
+
+// Test passes but this is an answer found on google and I'm not sure whats going on here.
```

Comment:
I would like to see you break down whats going on here and explain it!

Reply:
Sure, I'll take another look at it with fresh eyes and see if I can understand and explain it better. I'll post the explanation in a slack DM to you, because I'm not sure if I can still comment here once the PR is merged and closed.